src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java

```diff
@@ -448,6 +450,56 @@ public void onTick() {
         entityLabel.setFontRenderer(fontRenderer);
         frames.add(frame);
 
+        /*
+         * Potion Effects
+         */
+        frame = new Frame(getTheme(), new Stretcherlayout(1), "Potions");
+        frame.setCloseable(false);
+        frame.setPinnable(true);
+        Label potionsLabel = new Label("");
+        potionsLabel.addTickListener(new TickListener() {
+            Minecraft mc = Minecraft.getMinecraft();
+
+            @Override
+            public void onTick() {
+                potionsLabel.setText("");
+
+                if(mc.player == null){
+                    return;
+                }
+
+                // loop through potions and add them to the potions label
+                mc.player.getActivePotionMap().forEach((potion, potionEffect) -> {
+                    String name = I18n.format(potion.getName());
+                    String amplifier;
+                    switch (potionEffect.getAmplifier()) {
+                        case 0:
+                            amplifier = "I";
+                            break;
+                        case 1:
+                            amplifier = "II";
+                            break;
+                        default:
+                            amplifier = "?";
+                            break;
+                    }
+                    int secDuration = potionEffect.getDuration() / 20; // 20 ticks per sec, this may be incorrect in laggy servers
```

Comment:
is there no other day to do this?

Reply:
If there is a method to grab the tps from the server ad-hoc that I missed I'll be happy to remove the magic number. I didn't see any way to do this when scanning the minecraft connection / ServerWorld objects but I may have missed it.

Reply:
LagCompensator.INSTANCE.tickRate

Reply:
Whoops! I'll be sure to check the utility classes more closely next time. Thank you!

Reply:
It's all good lol. Not many people are familiar with the codebase. I try to document whatever's new, Xiaro does a better job at that